File: home/bin/pick.py
#!/usr/bin/env python

import re
import sys

# Ensure command line argument was provided
if len(sys.argv) < 2:
    print("Please provide a regex as a command line argument.")
    sys.exit(1)

# Get the regex from command line arguments
try:
    regex = re.compile(sys.argv[1])
except re.error as e:
    print("Invalid regex: " + str(e))
    sys.exit(1)

data = sys.stdin.read()

# Find matches and groups
matches = regex.finditer(data)

for match in matches:
    for group in match.groups():
        print(group)

File: home/bin/pick.mjs
#!/usr/bin/env node

const re = new RegExp(process.argv[2])

process.stdin.setEncoding('utf8')

process.stdin.on('data', function (data) {
  const a = String(data).match(re)
  process.stdout.write(a[1])
})

process.stdin.on('end', function () {
  process.exit()
})
